### ISSUE TITLE
complex JSON env variable behavior

### ISSUE THREAD

**Issue by grawk** [krakenjs/confit#42]
From experimenting I find that only strings at the base level of `process.env` can be used correctly in confit.

E.g. if my config.json is:

```
{
  "foo": {
    "bar": "baz"
  }
}
```

And I set this to the environment:

```
process.env.foo = {
  bar: 'bang'
}
```

In my resolved configuration I see:

```
foo: '[object Object]',
```

If I stringify the JSON, instead I see the following in resolved config:

```
foo: '{"bar": "bang"}'
```

I understand that env traditionally holds string values, but it would be nice if I could stringify a complex JSON object and confit would pull overrides out of that, like this:

```
process.env.foo = JSON.stringify({
  bar: 'bang'
});
```

resolves in config as:

```
foo: {
  bar: 'bang'
}
```

I have a use case that could be satisfied by such a feature. Do you think it is feasible to add it? Or is it something you'd accept a PR for?

**Comment by totherik** [comment]
Is this intended for creating globals programmatically, or passing them in from a true external env variable?

Also, one thing to be aware of is that confit won't pick up env variables that are created _after_ it is initialized.  It's not a live store. (Maybe it should be?)

**Comment by grawk** [comment]
Passing them in from an external env variable. Currently my module handles config in this way. It can be passed in via a stringified JSON environment variable, and the parsed environment variable is then merged over any existing configuration passed in via the module constructor. But I'm evaluating confit instead of my own homebrew method of handling config. I like the built in shortstop handlers and other API features. I'd have to work around the lack of this particular feature.

**Comment by grawk** [comment]
Created https://github.com/krakenjs/confit/pull/43 for this

**Comment by pvenkatakrishnan** [comment]
... looking into this.

**Comment by zetlen** [comment]
Rather than stringified JSON in arguments or environment variables, would it be simpler to use the [existing object traversal convention of confit](https://github.com/krakenjs/confit#config-api)?

Instead of this:

``` sh
$ node app.js --foo="{\"bar\":\"baz\"}"
```

You could do:

``` sh
$ node app.js --foo:bar=baz
```

Created https://github.com/krakenjs/confit/pull/52 to support this, using the existing object traversal code.

**Comment by shaunwarman** [comment]
It looks like this was resolved in #52 @zetlen ?

**Comment by zetlen** [comment]
@shaunwarman Sure does. Looks like it was always supported via long command arguments.
